sympy/polys/matrices/domainmatrix.py

```diff
@@ -21,6 +21,171 @@
 
 
 class DomainMatrix:
+    r"""
+
+        Module for the DomainMatrix class.
+
+        The idea behind DomainMatrix is to associate "domain" with matrix,
+        which could be found analogous to numpy arrays with "dtype".
+        In the DomainMatrix, each matrix has a domain such as ZZ(Integers) or QQ<sqrt(2)>[Algebraic Number Field].
```

Comment:
It would be helpful here to make a sphinx reference to
```
:py:class:`~.Domain`
```
That way someone reading the docs can click the link to find out more about domains.

Reply:
Should the `See Also` feature be added with a link to `domain` or just to replace `domain` with 
```
:py:class:`~.Domain` 
```

Reply:
I think that a see also would be good at the end of the docstring. It would be good here to make the first reference to "domain" a clickable link to the Domain class as well.

Reply:
Refences like ZZ can use `:ref:ZZ` I think.